ndcube/utils/wcs.py

```diff
@@ -140,75 +148,145 @@ def _wcs_slicer(wcs, missing_axis, item):
         Altered missing axis list.  Note the ordering has been reversed to reflect the data
         (numpy) axis ordering convention.
 
+    dropped_coords:
+        Coordinates which have been dropped in the slicing process is collected
+        in a dictionary called `dropped_coords`.
+
     """
+    # Force item to be WCS order if entered by user in numpy order.
+    if numpy_order is False:
+        # Convert item to a tuple is not already so we can handle it in a common way.
+        if isinstance(item, (int, np.int64, slice)):
+            item = (item)
+        # If item does not contain entries for all data axes, add empty slices for those axes.
+        n_add_axes = sum(np.invert(missing_axis)) - len(item)
+        item = tuple(list(item) + [slice(None)]*n_add_axes)
+        # Reverse order of item to be in WCS order.
+        item = item[::-1]
     # normal slice.
     item_checked = []
     if isinstance(item, slice):
         index = 0
-        # creating a new tuple of slice where if the axis is dead i.e missing
-        # then slice(0,1) added else slice(None, None, None) is appended and
+        # Creating a new tuple of slice where if the axis is dead, i.e. missing,
+        # then slice(0,1) added, else slice(None, None, None) is appended; and,
         # if the check of missing_axis gives that this is the index where it
         # needs to be appended then it gets appended there.
-        for _bool in missing_axis:
+        for i, _bool in enumerate(missing_axis):
             if not _bool:
-                if index is not 1:
+                if index != 1:
```

Comment:
I found this line a bit confusing, but not because of any changes you've made.  Could you change it to
```python
if index == 0:
```
?

Reply:
Weird... I don't see this in my version of the code base despite checking. I do see `if index is not len(item)` which I am changing to `if index != len(item)` in `wcs.py`, however.

Reply:
Try pulling the latest verion of ndcube from the main repo...